pkg/storage/store.go

```diff
@@ -35,13 +41,31 @@ func NewStore(
 	logger *zap.Logger,
 	config DBConfig,
 	ldClient *ld.LDClient,
+	dbIamFlag bool,
+	dbIamRoleArn string,
 ) (*Store, error) {
 	// LifecycleIDs are generated based on Eastern Time
 	tz, err := time.LoadLocation("America/New_York")
 	if err != nil {
 		return nil, err
 	}
 
+	var sess *session.Session
+	if dbIamFlag {
+		sess = session.Must(session.NewSession())
+	}
+
+	var creds *credentials.Credentials
```

Comment:
Can you move the `var sess` and `var creds` lines to just before line 80, so they're closer to where they're used? Also, I think you could move `sess = session.Must(session.NewSession())` before line 82 (that would remove the need for these two `if dbIamFlag` blocks.